underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -261,7 +276,91 @@ public boolean mkdirs(String path, boolean createParent) throws IOException {
 
   @Override
   public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
-    return true;
+    if (path == null) {
+      LOG.error("Attempting to create directory with a null path");
+      return false;
+    }
+    if (isDirectory(path)) {
+      return true;
+    }
+    if (isFile(path)) {
+      LOG.error("Cannot create directory {} because it is already a file.", path);
+      return false;
+    }
+    if (isRoot(path)) {
+      // nothing to do for root
+      return true;
+    }
+
+    if (!parentExists(path)) {
+      if (!options.getCreateParent()) {
+        LOG.error("Cannot create directory {} because parent does not exist", path);
+        return false;
+      }
+      final String parentKey = getParentKey(path);
+      // Recursively make the parent folders
+      if (!mkdirs(parentKey, true)) {
+        LOG.error("Unable to create parent directory", path);
+        return false;
+      }
+    }
+    return mkdirsInternal(path);
+  }
+
+  /**
+   * Creates a directory flagged file with the folder suffix.
+   *
+   * @param path the path to create a folder
+   * @return true if the operation was successful, false otherwise
+   */
+  private boolean mkdirsInternal(String path) {
+    try {
+      String keyAsFolder = addFolderSuffixIfNotPresent(
+          UnderFileSystemUtils.stripPrefixIfPresent(path, Constants.HEADER_SWIFT));
+      SwiftDirectClient.put(mAccess, keyAsFolder).close();
+      return true;
+    } catch (IOException e) {
+      LOG.error("Failed to create directory: {}", path, e);
+      return false;
+    }
+  }
+
+  /**
+   * Treating Swift as a file system, checks if the parent directory exists.
+   *
+   * @param path the path to check
+   * @return true if the parent exists or if the path is root, false otherwise
+   */
+  private boolean parentExists(String path) throws IOException {
+    final String parentKey = getParentKey(path);
+    return parentKey != null && isDirectory(parentKey);
+  }
+
+  /**
+   * @param path the path to get the parent of
+   * @return the parent path, or null if path is root
+   */
+  private String getParentKey(String path) {
+    // Root does not have a parent.
+    if (isRoot(path)) {
+      return null;
+    }
+    int separatorIndex = path.lastIndexOf(PATH_SEPARATOR);
+    if (separatorIndex < 0) {
```

Comment:
good catch. the path is of the form swift://root-container/path. so any expected path should have a PATH_SEPERATOR. i'll log an error message if that does not occur. or throw an exception instead?